src/algopy_testing/models/contract.py

```diff
@@ -42,24 +53,84 @@ def __init_subclass__(
         *,
         name: str | None = None,
         scratch_slots: (
-            algopy.UInt64 | tuple[int | algopy.UInt64, ...] | list[int | algopy.UInt64] | None
+            algopy.urange | tuple[int | algopy.urange, ...] | list[int | algopy.urange] | None
         ) = None,
         state_totals: StateTotals | None = None,
     ):
         cls._name = name or cls.__name__
         cls._scratch_slots = scratch_slots
         cls._state_totals = state_totals
 
+    def _get_local_states(self) -> dict[bytes, algopy.LocalState[Any]]:
+        import algopy
+
+        local_states = {
+            attribute._key.value: attribute
+            for _, attribute in vars(self).items()
+            if isinstance(attribute, algopy.LocalState)
+        }
+
+        return local_states
+
+    def _get_global_states(self) -> dict[bytes, algopy.GlobalState[Any]]:
+        import algopy
+
+        global_states = {}
+        for key, attribute in vars(self).items():
+            if not isinstance(attribute, algopy.LocalState) and not callable(attribute):
+                if isinstance(attribute, algopy.GlobalState):
+                    global_states[attribute._key.value] = attribute
+                else:
+                    global_states[key.encode()] = attribute
+
+        return global_states
+
+    def _count_state_types(
+        self, states: dict[bytes, algopy.GlobalState[Any]] | dict[bytes, algopy.LocalState[Any]]
+    ) -> tuple[int, int]:
+        import algopy
+
+        uint_count = sum(
+            1
+            for state in states.values()
+            if is_instance(
+                (
+                    state.type_
+                    if is_instance(state, (algopy.GlobalState | algopy.LocalState))
+                    else state
+                ),
+                algopy.UInt64
+                | algopy.arc4.UIntN
```

Comment:
These arc4 types should not be considered uint's for the purposes of this check, as they are actually encoded as a bytes value

Also should consider a native `bool` as a uint64 backed value

Reply:
Resolved, feel free to reopen if you think something else is missing